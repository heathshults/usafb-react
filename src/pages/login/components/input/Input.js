import React from 'react';
import PropTypes from 'prop-types';

import FontAwesome from 'react-fontawesome';

import './input.css';

const input = props => (
  <div className="row form-group">
    <div className="col-md-12 input-group">
      <label htmlFor="txtEmail" className="login-page__email-label sr-only">
        {props.placeholder}
      </label>
      <div className="col input-wrapper">
        <FontAwesome name={props.icon} className="fa-layer" />
        <input
          type={props.inputType}
          className="form-control form-control-theme"
          id="txtEmail"
          placeholder={props.placeholder}
          value={props.value}
          onChange={props.onChange}
        />
      </div>
    </div>
  </div>
);

input.propTypes = {
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  icon: PropTypes.string.isRequired,
  placeholder: PropTypes.string.isRequired,
  inputType: PropTypes.string
};

input.defaultProps = {
  inputType: 'text'
};

export default input;
