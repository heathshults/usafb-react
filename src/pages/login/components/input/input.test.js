import React from 'react';
import Input from './Input';
import renderer from 'react-test-renderer';

test('Read input changes ', () => {
  const email = '';

  const updateEmail = (event) => {
    email = event.target.value;
  }

  const component = renderer.create(
    <Input
      icon="envelope"
      placeholder="Email"
      value={email}
      onChange={updateEmail}
    />,
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  console.log(tree.props);

  // // manually trigger the callback
  // tree.props.onMouseEnter();
  // // re-rendering
  // tree = component.toJSON();
  // expect(tree).toMatchSnapshot();

  // // manually trigger the callback
  // tree.props.onMouseLeave();
  // // re-rendering
  // tree = component.toJSON();
  // expect(tree).toMatchSnapshot();
});