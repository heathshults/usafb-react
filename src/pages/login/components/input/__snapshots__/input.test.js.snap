// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Read input changes  1`] = `
<div
  className="row form-group"
>
  <div
    className="col-md-12 input-group"
  >
    <label
      className="login-page__input-label sr-only"
      htmlFor="txtEmail"
    >
      Email
    </label>
    <div
      className="col input-wrapper"
    >
      <span
        aria-hidden={true}
        className="fa fa-envelope fa-layer"
      />
      <input
        className="form-control form-control-theme"
        id="txtEmail"
        onChange={[Function]}
        placeholder="Email"
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;
